### PR TITLE
Breaking change backend

### PR DESCRIPTION
**Pull request by iamthuypham**
* [x] Fix several api routes
* [x] Add error handler
* [x] Add several global environment variables

# Description

[Trello Card](paste trello card url here)

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status
- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: client/src/components/premium/PremiumView.js [hunk 1]
```diff
@@ -102,130 +118,172 @@ class PremiumView extends Component {
...
-                        currentSub="Monthly"
-                        premium={this.state.premium}
-                      />
-                    </div>
-                  </>
-                )
-              )}
+                <>
+                  <div className="col-md-6">
+                    <PremiumCard
```
Comment (ckopecky) [inline client/src/components/premium/PremiumView.js:254]: `currentUser={this.state}
onCompleteSuccessfulPayment={ this.handleCompleteSuccessfulPayment } 
`

??

File: client/src/components/premium/PremiumView.js [hunk 2]
```diff
@@ -102,130 +118,172 @@ class PremiumView extends Component {
...
+                        'TLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
+                      }
+                      totalCents={999}
+                      displayNone="displayNone"
+                      premium={this.state.premium}
+                    />
+                  </div>
 
+                  <div className="col-md-6">
+                    <PremiumCard
```
Comment (ckopecky) [inline client/src/components/premium/PremiumView.js:268]: `currentUser={this.state} onCompleteSuccessfulPayment={ this.handleCompleteSuccessfulPayment }`

??

File: client/src/services/resourceURLs.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const theMovieDbUrl = `${process.env.REACT_APP_THE_MOVIE_DB_URL}`;
+export const tmdbUrl = `${process.env.REACT_APP_TMDB_URL}`;
+export const placeholderUrl = `${process.env.REACT_APP_PLACEHOLDER_URL}`;
+export const viaPlaceholderUrl = `${process.env.REACT_APP_VIA_PLACEHOLDER_URL}`;
+export const unsplashUrl = `${process.env.REACT_APP_UNPLASH_URL}`;
```
Comment (ckopecky) [inline client/src/services/resourceURLs.js]: export const unsplashUrl = {..........UN**S**PLASH...};

File: server/api/authentication/authRouter.js [hunk 4]
```diff
@@ -53,20 +47,34 @@ router.get('/logout', (req, res) => {
...
-    res.redirect(process.env.REDIRECT_URI_PROD);
-  } else res.redirect(process.env.REDIRECT_URI_DEV);
-  
+  // load the latest data of this user
+  if (user) {
+    res.status(200).send(req.user);
+  }
+
+  // load data from request
+  const { id, name, email, username } = req.user;
```
Comment (ckopecky) [inline server/api/authentication/authRouter.js]: `  const { id, name, email, username, photo } = req.user;
`

File: server/api/authentication/authRouter.js [hunk 5]
```diff
@@ -53,20 +47,34 @@ router.get('/logout', (req, res) => {
...
+  res.status(200).send({ id, name, email, username });
```
Comment (ckopecky) [inline server/api/authentication/authRouter.js]: `  res.status(200).send({ id, name, email, username, photo });`

File: server/api/payments/paymentRouter.js [hunk 6]
```diff
@@ -10,131 +10,109 @@ const router = require('express').Router();
...
-            .subscriptions
-            .data[0]
-            .items
-            .data[0]
-            .plan
-            .active    
-        })
-      }
+  stripe.customers.retrieve(stripeid, function(err, customer) {
+    if (customer.deleted) {
```
Comment (ckopecky) [inline server/api/payments/paymentRouter.js]: should we have a condition that asks if customer exists first like you did above?

File: server/api/users/usersRouter.js [hunk 7]
```diff
@@ -8,16 +8,24 @@ const router = require('express').Router();
...
+  usersDb.getUsers().then(users =>
+    res
+      .status(200)
+      .send(users)
+      .catch(error => res.status(500).send(error))
+  );
 });
 
 // GET request that gets a user by id
 router.get('/users/:id', async (req, res) => {
```
Comment (ckopecky) [inline server/api/users/usersRouter.js]: no `async` needed?

**Review by ckopecky — CHANGES_REQUESTED** [review]

**Comment by ckopecky** [comment]
This is amazing, Thuy. Thank you for the once over on our code. I want to have Ari give it a once over too.

File: client/src/components/premium/PremiumView.js (hunk 1 above)
Comment (iamthuypham) [reply client/src/components/premium/PremiumView.js:254]: In `componentDidMount`, I fetch the current user and stripe id (if any), store in state, then I pass it down to `PremiumCard` and `PayButton` to use those data.


In `PayButton`, when payment is successful, it notifies the `PremiumView` to update its' state.


Parent and Child component should get state from one place to ensure integrity.

File: client/src/components/premium/PremiumView.js (hunk 2 above)
Comment (iamthuypham) [reply client/src/components/premium/PremiumView.js:268]: The same in another comment.

File: client/src/services/resourceURLs.js (hunk 3 above)
Comment (iamthuypham) [reply client/src/services/resourceURLs.js]: 🤔 Did I spell something wrong ?

File: server/api/payments/paymentRouter.js (hunk 6 above)
Comment (iamthuypham) [reply server/api/payments/paymentRouter.js]: I missed this one. This route has not used anywhere yet, that is reason why I did not see the potential error.

But yea. It should.

File: client/src/components/premium/PremiumView.js (hunk 1 above)
Comment (ari-mercado) [reply client/src/components/premium/PremiumView.js:254]: I love this. It's the perfect solution to the problem I've had this week of not being able to update the state after the payment went through.

File: client/src/services/resourceURLs.js (hunk 3 above)
Comment (ari-mercado) [reply client/src/services/resourceURLs.js]: I believe there's an "S" missing between the "N" and the "P" in `${process.env.REACT_APP_UNPLASH_URL}`. However, I am not sure how you spelled it in the env file.

Comment (ckopecky) [reply client/src/services/resourceURLs.js]: Line 5. It's Unsplash and not unplash, right? Or I guess it didn't matter as long as it matches whatever isin env file!

Comment (iamthuypham) [reply client/src/services/resourceURLs.js]: I will change that

File: client/src/components/movieReviews/MovieRev_wireframe.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import './MovieRev.css';
 import { Container, Row, Col, Button } from 'reactstrap';
+import { placeholderUrl } from '../../services/resourceURLs';
```
Comment (ckopecky) [inline client/src/components/movieReviews/MovieRev_wireframe.js]: `import { placeholderUrl, viaPlaceholderUrl } from '../../services/resourceURLs';`

File: client/src/components/navbar/Navigation.js [hunk 9]
```diff
@@ -19,30 +19,28 @@ import { googleLogin, logout } from '../../services/authURLs';
...
-      photo:''
-
+      photo: ''
     };
     this.toggle = this.toggle.bind(this);
-
   }
 
   componentDidMount = async () => {
     const res = await axios.get(currentUser, {
```
Comment (ckopecky) [inline client/src/components/navbar/Navigation.js:37]: `const res = await axios.get(currentUser, {`
should be: 
    `const res = await axios.get(users, {`

to reflect the change in var name you made above.

**Review by ckopecky — CHANGES_REQUESTED** [review]
Just two little minor things. I pulled your code and my linter caught just two things!

File: client/src/components/movieReviews/MovieRev_wireframe.js (hunk 8 above)
Comment (iamthuypham) [reply client/src/components/movieReviews/MovieRev_wireframe.js]: Good catch. We should install eslint for you guys next pr!

File: client/src/components/navbar/Navigation.js (hunk 9 above)
Comment (iamthuypham) [reply client/src/components/navbar/Navigation.js:37]: Oh, it should the other way around. But yea. `import { currentUser } from ...`.

Comment (iamthuypham) [reply client/src/components/navbar/Navigation.js:37]: We dont use Redux so, eventually, we want to make request `currentUser` at top level component, and then pass user result down to whichever children components (such as `Navigation`).

Only the child that too nested can make request again.

**Review by ckopecky — APPROVED** [review]

**Review by ari7946 — APPROVED** [review]

**Pull request merged**